server/ssg.ts

```diff
@@ -1,14 +1,30 @@
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import superjson from "superjson";
+
 import prisma from "@lib/prisma";
 
 import { createSSGHelpers } from "@trpc/react/ssg";
 
 import { appRouter } from "./routers/_app";
 
-export const ssg = createSSGHelpers({
-  router: appRouter,
-  ctx: {
-    prisma,
-    session: null,
-    user: null,
-  },
-});
+export async function ssgInit(opts: { locale: string }) {
+  const { locale } = opts;
+  const _i18n = await serverSideTranslations(locale, ["common"]);
+
+  const ssg = createSSGHelpers({
+    router: appRouter,
+    transformer: superjson,
+    ctx: {
+      prisma,
+      session: null,
+      user: null,
+      locale,
+      i18n: _i18n,
+    },
+  });
+
+  // always preload "viewer.i18n"
+  await ssg.fetchQuery("viewer.i18n");
```

Comment:
This makes sure that the first call to `useQuery(["viewer.i18n"])` always will be pre-populated